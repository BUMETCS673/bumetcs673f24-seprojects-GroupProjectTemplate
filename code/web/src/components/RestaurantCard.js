import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
// src/components/RestaurantCard.tsx
import { Link } from 'react-router-dom';
function RestaurantCard({ restaurant }) {
    return (_jsxs("div", { className: "bg-white rounded-lg shadow-md p-6", children: [_jsx("h2", { className: "text-2xl font-semibold mb-2", children: restaurant.name }), _jsx("p", { className: "text-gray-600 mb-2", children: restaurant.address }), _jsxs("p", { className: "text-gray-600 mb-4", children: ["Phone: ", restaurant.phone] }), _jsxs("div", { className: "flex items-center mb-4", children: [_jsx("span", { className: "text-yellow-500 mr-2", children: "Overall Rating:" }), _jsx("span", { className: "font-bold", children: restaurant.overall_rating.toFixed(1) })] }), _jsxs("div", { className: "flex space-x-4 mb-4", children: [_jsxs("div", { className: "flex items-center", children: [_jsx("span", { className: "text-green-500 mr-1", children: "Vegan:" }), _jsx("span", { children: restaurant.vegan_rating.toFixed(1) })] }), _jsxs("div", { className: "flex items-center", children: [_jsx("span", { className: "text-blue-500 mr-1", children: "Gluten-Free:" }), _jsx("span", { children: restaurant.gluten_free_rating.toFixed(1) })] }), _jsxs("div", { className: "flex items-center", children: [_jsx("span", { className: "text-purple-500 mr-1", children: "Vegetarian:" }), _jsx("span", { children: restaurant.vegetarian_rating.toFixed(1) })] })] }), _jsx(Link, { to: `/restaurant/${restaurant.restaurant_id}`, children: _jsx("button", { className: "bg-red-500 text-white px-4 py-2 rounded-md hover:bg-red-600", children: "View Details" }) })] }));
}
export default RestaurantCard;
