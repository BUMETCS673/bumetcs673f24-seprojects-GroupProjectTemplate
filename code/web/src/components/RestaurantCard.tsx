// src/components/RestaurantCard.tsx
import { Link } from 'react-router-dom';

interface Restaurant {
  restaurant_id: number;
  name: string;
  address: string;
  phone: string;
  overall_rating: number;
  vegan_rating: number;
  gluten_free_rating: number;
  vegetarian_rating: number;
}

interface RestaurantCardProps {
  restaurant: Restaurant;
}

function RestaurantCard({ restaurant }: RestaurantCardProps) {
  return (
    <div className="bg-white rounded-lg shadow-md p-6">
      <h2 className="text-2xl font-semibold mb-2">{restaurant.name}</h2>
      <p className="text-gray-600 mb-2">{restaurant.address}</p>
      <p className="text-gray-600 mb-4">Phone: {restaurant.phone}</p>
      <div className="flex items-center mb-4">
        <span className="text-yellow-500 mr-2">Overall Rating:</span>
        <span className="font-bold">{restaurant.overall_rating.toFixed(1)}</span>
      </div>
      <div className="flex space-x-4 mb-4">
        <div className="flex items-center">
          <span className="text-green-500 mr-1">Vegan:</span>
          <span>{restaurant.vegan_rating.toFixed(1)}</span>
        </div>
        <div className="flex items-center">
          <span className="text-blue-500 mr-1">Gluten-Free:</span>
          <span>{restaurant.gluten_free_rating.toFixed(1)}</span>
        </div>
        <div className="flex items-center">
          <span className="text-purple-500 mr-1">Vegetarian:</span>
          <span>{restaurant.vegetarian_rating.toFixed(1)}</span>
        </div>
      </div>
      <Link to={`/restaurant/${restaurant.restaurant_id}`}>
        <button className="bg-red-500 text-white px-4 py-2 rounded-md hover:bg-red-600">
          View Details
        </button>
      </Link>
    </div>
  );
}

export default RestaurantCard;
