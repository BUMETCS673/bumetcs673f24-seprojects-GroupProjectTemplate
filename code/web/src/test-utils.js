import { jsx as _jsx } from "react/jsx-runtime";
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
const AllTheProviders = ({ children }) => {
    return (_jsx(Router, { children: children }));
};
const customRender = (ui, options) => render(ui, Object.assign({ wrapper: AllTheProviders }, options));
export * from '@testing-library/react';
export { customRender as render };
