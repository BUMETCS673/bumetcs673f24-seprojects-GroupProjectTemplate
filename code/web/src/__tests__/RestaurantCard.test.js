import { jsx as _jsx } from "react/jsx-runtime";
import { render } from '../test-utils'; // Adjust this path if needed
import '@testing-library/jest-dom';
import RestaurantCard from '../../src/components/RestaurantCard';
describe('RestaurantCard Component', () => {
    const restaurant = {
        restaurant_id: 123,
        name: 'Test Restaurant',
        address: '123 Test St, Test City',
        phone: '555-1234',
        overall_rating: 4.25,
        vegan_rating: 4.1,
        gluten_free_rating: 3.7,
        vegetarian_rating: 4.9,
    };
    it('renders the restaurant name, address, and phone', () => {
        const { getByText } = render(_jsx(RestaurantCard, { restaurant: restaurant }));
        expect(getByText('Test Restaurant')).toBeInTheDocument();
        expect(getByText('123 Test St, Test City')).toBeInTheDocument();
        expect(getByText('Phone: 555-1234')).toBeInTheDocument();
    });
    it('renders ratings with correct formatting', () => {
        const { getByText } = render(_jsx(RestaurantCard, { restaurant: restaurant }));
        // Overall rating
        expect(getByText('Overall Rating:')).toBeInTheDocument();
        expect(getByText('4.3')).toBeInTheDocument(); // 4.25.toFixed(1) -> '4.3'
        // Vegan rating
        expect(getByText('Vegan:')).toBeInTheDocument();
        expect(getByText('4.1')).toBeInTheDocument();
        // Gluten-Free rating
        expect(getByText('Gluten-Free:')).toBeInTheDocument();
        expect(getByText('3.7')).toBeInTheDocument();
        // Vegetarian rating
        expect(getByText('Vegetarian:')).toBeInTheDocument();
        expect(getByText('4.9')).toBeInTheDocument();
    });
    it('renders a "View Details" button linking to the correct route', () => {
        const { getByRole } = render(_jsx(RestaurantCard, { restaurant: restaurant }));
        const button = getByRole('button', { name: /view details/i });
        expect(button).toBeInTheDocument();
        // The button is wrapped by a Link
        const link = button.closest('a');
        expect(link).toHaveAttribute('href', '/restaurant/123');
    });
});
