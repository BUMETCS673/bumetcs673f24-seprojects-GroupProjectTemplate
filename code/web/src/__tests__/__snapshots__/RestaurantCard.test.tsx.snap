// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow-md p-6"
  >
    <h2
      class="text-2xl font-semibold mb-2"
    >
      Test Restaurant
    </h2>
    <p
      class="text-gray-600 mb-2"
    >
      123 Test St, Test City
    </p>
    <p
      class="text-gray-600 mb-4"
    >
      Phone: 555-1234
    </p>
    <div
      class="flex items-center mb-4"
    >
      <span
        class="text-yellow-500 mr-2"
      >
        Overall Rating:
      </span>
      <span
        class="font-bold"
      >
        4.3
      </span>
    </div>
    <div
      class="flex space-x-4 mb-4"
    >
      <div
        class="flex items-center"
      >
        <span
          class="text-green-500 mr-1"
        >
          Vegan:
        </span>
        <span>
          4.1
        </span>
      </div>
      <div
        class="flex items-center"
      >
        <span
          class="text-blue-500 mr-1"
        >
          Gluten-Free:
        </span>
        <span>
          3.7
        </span>
      </div>
      <div
        class="flex items-center"
      >
        <span
          class="text-purple-500 mr-1"
        >
          Vegetarian:
        </span>
        <span>
          4.9
        </span>
      </div>
    </div>
    <a
      data-discover="true"
      href="/restaurant/123"
    >
      <button
        class="bg-red-500 text-white px-4 py-2 rounded-md hover:bg-red-600"
      >
        View Details
      </button>
    </a>
  </div>
</DocumentFragment>
`;
